File: nomad-samples/js/account-updates/account/update-user.js
import * as prjConstants from "../constants/project-constants.js";
import apiExceptionHandler from "../exceptions/api-exception-handler.js";

export default async function updateUser(AUTH_TOKEN, ADDRESS1, ADDRESS2, CITY, COUNTRY, FIRST_NAME, LAST_NAME, 
    ORGANIZATION, PHONE_NUMBER, PHONE_EXT, POSTAL_CODE, STATE) 
{
    // Create header for the request
    const HEADERS = new Headers();
    HEADERS.append("Content-Type", "application/json");
    HEADERS.append("Authorization", `Bearer ${AUTH_TOKEN}`);
  
  	const BODY = {
        address: ADDRESS1,
        address2: ADDRESS2,
        city: CITY,
        state: STATE,
        country: COUNTRY,
        firstName: FIRST_NAME,
        lastName: LAST_NAME,
        phoneExt: PHONE_EXT,
        phone: PHONE_NUMBER,
        postalCode: POSTAL_CODE,
        organization: ORGANIZATION
    };
  
    // Post
    const RESPONSE = await fetch(`${prjConstants.PORTAL_API_URL}/account/user`, {
        method: "PUT",
        headers: HEADERS,
        body: JSON.stringify(BODY)
    }).catch((exception) => {
        throw exception;
    });

    // Check for success
    if (RESPONSE.ok) {
        // Get the JSON from the response
        const INFO = await RESPONSE.json();

        return INFO;
    }
		
  	// There was an error
    await apiExceptionHandler(RESPONSE, "Update user failed");
}

File: nomad-samples/python/Account Updates/account/update_user.py
from constants.project_constants import *
from exceptions.api_exception_handler import *

import json, requests

def update_user(AUTH_TOKEN: str) -> dict:
    if (not AUTH_TOKEN):
        raise Exception("Authentication Token: The authentication token is invalid")
  
    API_URL = PORTAL_API_URL + "/account"
  
    # Create header for the request
    HEADERS = {
        "Content-Type": "application/json"
    }

    # Build the payload body
    BODY = {
        "id": "990a1ebc-3344-41fc-a331-4009b3773229",
        "email": "d.fletcher@example.net",
        "firstName": "Deanna ",
        "lastName": "Fletcher",
        "mobilePhone": "949-555-4545"
    }
    
    try:
        RESPONSE = requests.post(API_URL, headers=HEADERS, data=json.dumps(BODY))
        if RESPONSE.status_code != 200:
            raise Exception("Response returned " + str(RESPONSE.status_code))

        return json.loads(RESPONSE.text)
    except:
        api_exception_handler(RESPONSE, "Update user failed")